### PR TITLE
Project/afw table/ishasan

### PR DESCRIPTION
**Pull request by ih64**
added a few new methods that help users navigate large schemas. also included some clarification on distinction between schema keys and python dictionary keys, as they were interchangeably used in the previous notebook. Once this is merged in #51 can be closed

### CODE REVIEW

**Comment by ih64** [comment]
also fixed up the notebook so it will run with version 17 of the stack

File: Basics/afw_table_guided_tour.ipynb [hunk 1]
```diff
@@ -392,7 +420,7 @@
    "metadata": {},
    "source": [
     "### Data access\n",
-    "If you know the path to your source catalog, there is a quick way to read it in. However, it is often more powerful to the 'data butler' to fetch data for you. The butler knows about camera geometry, sensor characteristics, where data are located, and so forth. Having this anciliary information on hand is often very useful. For completeness we will demonstrate both ways of reading in a source catalog, with the note that it is largely considered better practice to use the data butler. \n",
+    "If you know the path to your source catalog, there is a quick way to read it in. However, it is often more powerful to use the 'data butler' to fetch data for you. The butler knows about camera geometry, sensor characteristics, where data are located, and so forth. Having this anciliary information on hand is often very useful. For completeness we will demonstrate both ways of reading in a source catalog, with the note that it is largely considered better practice to use the data butler. \n",
```
Comment (kadrlica) [inline Basics/afw_table_guided_tour.ipynb]: Do we have a tutorial on the butler? Could be useful to add a reference here.

File: Basics/afw_table_guided_tour.ipynb [hunk 2]
```diff
@@ -469,7 +513,27 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "If we are just intested in the field names, we can do this:"
+    "Schemas in source catalogs include fields for measurements, and the associated flag fields for those measurements.  To tally how many fields, flag fields, and non flag fields are contained, we can use several count methods."
```
Comment (kadrlica) [inline Basics/afw_table_guided_tour.ipynb]: should be "non-flag"

File: Basics/afw_table_guided_tour.ipynb [hunk 3]
```diff
@@ -1102,10 +1208,11 @@
    "metadata": {},
    "source": [
     "We can see that some of the matches have very similar magnitude residuals. Look at the 6th and 7th to last row, for example:\n",
+    "\n",
```
Comment (kadrlica) [inline Basics/afw_table_guided_tour.ipynb]: This is probably fragile and you might want to think about a way to avoid hard-coding the values.

**Pull request merged**